lib/stringlib/lua/matching.lua

```diff
@@ -120,6 +123,35 @@ do
     --> =
     --> =b
     --> =
+
+    -- Lua 5.4 introduces an optional third argument to string.gmatch,
+    -- specifying where to start the search.
+
+    for w in string.gmatch(s, "%a+", 8) do
+        print(w)
+    end
+    --> =orld
+    --> =from
+    --> =Lua
```

Comment:
The results of these tests seem wrong to me: 

I think this one should produce:
```
rld
to
Lua
```

Reply:
`s` is the string "hello world from Lua", so I think my test is correct (same for the other remarks). What do you think `s` is?

Reply:
Ah that makes sense, in the diff I saw a line `local s = "from=world, to=Lua"` right above these tests, thought that was the `s` that's used, my bad :+1: